samcli/lib/sync/flows/layer_sync_flow.py

```diff
@@ -258,6 +259,30 @@ def _get_dependent_functions(self) -> List[Function]:
         return dependent_functions
 
 
+class LayerSyncFlowSkipBuildDirectory(LayerSyncFlow):
+    """
+    LayerSyncFlow special implementation that will skip build step and zip contents of CodeUri
+    """
+
+    def gather_resources(self) -> None:
+        zip_file_path = os.path.join(tempfile.gettempdir(), f"data-{uuid.uuid4().hex}")
```

Comment:
nit: Should we try use `Pathlib` where we can instead of `os.path`?

Reply:
We could, but I just wanted to follow what we already have in this file. We could do it in another PR for refactoring purposes.